src/test/java/com/knighttodo/knighttodo/integration/RoutineResourceIntegrationTest.java

```diff
@@ -0,0 +1,167 @@
+package com.knighttodo.knighttodo.integration;
+
+import static com.knighttodo.knighttodo.Constants.API_BASE_BLOCKS;
+import static com.knighttodo.knighttodo.Constants.API_BASE_ROUTINES;
+import static com.knighttodo.knighttodo.TestConstants.buildDeleteRoutineByIdUrl;
+import static com.knighttodo.knighttodo.TestConstants.buildGetRoutineByIdUrl;
+import static com.knighttodo.knighttodo.TestConstants.buildJsonPathToHardness;
+import static com.knighttodo.knighttodo.TestConstants.buildJsonPathToId;
+import static com.knighttodo.knighttodo.TestConstants.buildJsonPathToLength;
+import static com.knighttodo.knighttodo.TestConstants.buildJsonPathToName;
+import static com.knighttodo.knighttodo.TestConstants.buildJsonPathToReadyName;
+import static com.knighttodo.knighttodo.TestConstants.buildJsonPathToScariness;
+import static com.knighttodo.knighttodo.TestConstants.buildJsonPathToTemplateIdName;
+import static com.knighttodo.knighttodo.TestConstants.buildJsonPathToTodoBlockId;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.knighttodo.knighttodo.factories.RoutineFactory;
+import com.knighttodo.knighttodo.factories.TodoBlockFactory;
+import com.knighttodo.knighttodo.gateway.privatedb.repository.RoutineRepository;
+import com.knighttodo.knighttodo.gateway.privatedb.repository.TodoBlockRepository;
+import com.knighttodo.knighttodo.gateway.privatedb.representation.Routine;
+import com.knighttodo.knighttodo.gateway.privatedb.representation.TodoBlock;
+import com.knighttodo.knighttodo.rest.dto.routine.request.CreateRoutineRequestDto;
+import com.knighttodo.knighttodo.rest.dto.routine.request.UpdateRoutineRequestDto;
+import com.knighttodo.knighttodo.rest.dto.routine.response.RoutineResponseDto;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.transaction.annotation.Transactional;
+
+@ExtendWith(SpringExtension.class)
```

Comment:
I think it should work without this annotation

Reply:
yes, you're right. will delete it. Thanks.